### PR TITLE
Add fluent speech commands

### PR DESCRIPTION
**Pull request by carolineechen**

### CODE REVIEW

File: docs/source/refs.bib [hunk 1]
```diff
@@ -359,3 +359,16 @@ @misc{YesNo
   title="YesNo",
   url="http://www.openslr.org/1/"
 }
+@inproceedings{fluent,
+  author    = {Loren Lugosch and
+               Mirco Ravanelli and
+               Patrick Ignoto and
+               Vikrant Singh Tomar and
+               Yoshua Bengio},
```
Comment (nateanl) [inline docs/source/refs.bib:367]: Making them into one line looks better?

File: docs/source/refs.bib [hunk 2]
```diff
@@ -359,3 +359,16 @@ @misc{YesNo
...
+  editor    = {Gernot Kubin and
+               Zdravko Kacic},
```
Comment (nateanl) [inline docs/source/refs.bib:369]: same here

File: torchaudio/datasets/fluentcommands.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+FOLDER_IN_ARCHIVE = "fluent_speech_commands_dataset"
+
+
+class FluentSpeechCommands(Dataset):
+    """Create *Fluent Speech Commands* [:footcite:`fluent`] Dataset
+
+    Args:
+        root (str of Path): Path to the directory where the dataset is found.
+        subset (str): subset of the dataset to use. Options: ["train", "valid", "test"].
```
Comment (nateanl) [inline torchaudio/datasets/fluentcommands.py:17]:
```suggestion
        subset (str, optional): subset of the dataset to use. Options: [`"train"`, `"valid"`, `"test"`]. (Default: `"train"`)
```

File: torchaudio/datasets/fluentcommands.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            (Default: ``"train"``)
+    """
+
+    def __init__(self, root: Union[str, Path], subset: str = "train"):
+        assert subset in ["train", "valid", "test"]
```
Comment (nateanl) [inline torchaudio/datasets/fluentcommands.py:22]: Add an exception message after assert makes clearer to users.

File: test/torchaudio_unittest/datasets/fluentcommands_test.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+from torchaudio.datasets import fluentcommands
+from torchaudio_unittest.common_utils import (
+    get_whitenoise,
+    save_wav,
+    TempDirMixin,
+    TorchaudioTestCase,
+)
+
+HEADER = ["", "path", "speakerId", "transcription", "action", "object", "location"]
```
Comment (nateanl) [inline test/torchaudio_unittest/datasets/fluentcommands_test.py:15]: Is the `speakerId` from original csv? Or can we change it to
```suggestion
HEADER = ["", "path", "speaker_id", "transcription", "action", "object", "location"]
```

File: torchaudio/datasets/fluentcommands.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import csv
+import os
+from pathlib import Path
+from typing import Union
+
+import torchaudio
+from torch.utils.data import Dataset
+
+FOLDER_IN_ARCHIVE = "fluent_speech_commands_dataset"
```
Comment (nateanl) [inline torchaudio/datasets/fluentcommands.py:9]: shall we add prefix underscore?
```suggestion
_FOLDER_IN_ARCHIVE = "fluent_speech_commands_dataset"
```

**Review by nateanl — APPROVED** [review]
LGTM. Just some nits to address.

File: test/torchaudio_unittest/datasets/fluentcommands_test.py (hunk 5 above)
Comment (carolineechen) [reply test/torchaudio_unittest/datasets/fluentcommands_test.py:15]: yea it is from the original csv

File: torchaudio/datasets/fluentcommands.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import csv
+import os
+from pathlib import Path
+from typing import Union
+
+import torchaudio
+from torch.utils.data import Dataset
+
+_FOLDER_IN_ARCHIVE = "fluent_speech_commands_dataset"
```
Comment (mthrok) [inline torchaudio/datasets/fluentcommands.py:9]: Why should this be global variable? It is only used once, and it is not expected to be changed.

**Pull request closed without merging**